pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+        else:
+            control_base_1 = operation1.control_base
+
+        target_wires_1 = qml.wires.Wires(
+            [w for w in operation1.wires if w not in operation1.control_wires]
+        )
+
+        control_target = intersection(operation1.control_wires, operation2.wires)
+        target_target = intersection(target_wires_1, operation2.wires)
+
+        # Case 3.1: control and target 1 overlap with target 2
+        if control_target and target_target:
+            return bool(commutation_map[control_base_1][position[operation2.name]]) and bool(
+                commutation_map["ctrl"][position[operation2.name]]
+            )
+
+        # Case 3.2: control operation 1 overlap with target 2
+        if control_target and not target_target:
+            return bool(commutation_map["ctrl"][position[operation2.name]])
+
+        # Case 3.3: target 1 overlaps with target 2
+        if not control_target and target_target:
+            return bool(commutation_map[control_base_1][position[operation2.name]])
+
+    # Case 4: only operation 2 is controlled
+    elif control_base.get(operation2.name):
+        if control_base.get(operation2.name) != "ControlledOperation":
+            control_base_2 = control_base.get(operation2.name)
+        else:
+            control_base_2 = operation2.control_base
+
+        target_wires_2 = qml.wires.Wires(
+            [w for w in operation2.wires if w not in operation2.control_wires]
+        )
+
+        target_control = intersection(operation1.wires, operation2.control_wires)
+        target_target = intersection(operation1.wires, target_wires_2)
+
+        # Case 4.1: control and target 2 overlap with target 1
+        if target_control and target_target:
+            return bool(commutation_map[operation1.name][position[control_base_2]]) and bool(
+                commutation_map[operation1.name][position[control_base_2]]
+            )
+
+        # Case 4.2: control operation 2 overlap with target 1
+        if target_control and not target_target:
+            return bool(commutation_map[operation1.name][position["ctrl"]])
+
+        # Case 4.3: target 1 overlaps with target 2
+        if not target_control and target_target:
+            return bool(commutation_map[operation1.name][position[control_base_2]])
+
+    # Case 5: no controlled operations
+    # Case 5.1: no controlled operations we simply check the commutation table
+    return bool(commutation_map[operation1.name][position[operation2.name]])
+
+
+def _merge_no_duplicates(*iterables):
+    """Merge K list without duplicate using python heapq ordered merging.
+
+    Args:
+        *iterables: A list of k sorted lists
+
+    Yields:
+        Iterator: List from the merging of the k ones (without duplicates)
+    """
+    last = object()
+    for val in heapq.merge(*iterables):
+        if val != last:
+            last = val
+            yield val
+
+
+class CommutationDAGNode:
```

Comment:
This doesn't appear in the documentation, but I'm not sure if that is intentional or not :)